main.cpp

```diff
@@ -211,6 +215,31 @@ static bool closeUsciPipes(Emulator* const emu)
     return true;
 }
 
+static bool openDioPipes(Emulator* const emu)
+{
+    if (emu->port1_output_pipe_name != NULL)
+    {
+        emu->port1_output_pipe = fopen(emu->port1_output_pipe_name, "wb");
+        if (emu->port1_output_pipe == NULL)
+        {
+            print_console(emu, "Cannot open DIGITAL I.O PORT 1 output pipe\n");
+            return false;
+        }
+        emu->port1_output_pipe_fd = fileno(emu->port1_output_pipe);
+    }
+    return true;
+}
+
+static bool closeDioPipes(Emulator* const emu)
+{
+    if (emu->port1_output_pipe != NULL)
+    {
+        fclose(emu->port1_output_pipe);
+        emu->port1_output_pipe = NULL;
```

Comment:
`_fd` = ???

mkay, I see it's a common pattern in this file to leave fd of closed file untouched.

Reply:
fd is derived from the opened file, which is now getting closed;the condition checked in other places to use the pipe/pipe_fd is pipe != NULL

Reply:
But ok, added cleaning up fd, just in case